### PR TITLE
[MSA] Encourage MSA logins on NuGet.org

### PR DESCRIPTION
**Pull request by shishirx34**
This PR makes UI changes for encouraging users to login using MSA.
- Moves the login/register using NuGet.org to the second page. 
- Adds telemetry for register accounts.
- Removes register link from the header.

TODO:
- [x] Need to fix the functional tests that will be broken by this change.

Snapshots:

Main Logon Page:
image

Logon with NuGet page:
image

### CODE REVIEW

**Comment by anangaur** [comment]
Can this be at the center of the white space between the header (blue part) and footer (other blue part)? I think that will look better.

image

**Comment by anangaur** [comment]
But do not block on this one. We can take this change later.

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -84,9 +84,9 @@
     <add key="Auth.LocalUser.Enabled" value="true" />
     <add key="Auth.ApiKey.Enabled" value="true" />
     
-    <add key="Auth.MicrosoftAccount.Enabled" value="false" />
-    <add key="Auth.MicrosoftAccount.ClientId" value="" />
-    <add key="Auth.MicrosoftAccount.ClientSecret" value="" />
+    <add key="Auth.MicrosoftAccount.Enabled" value="true" />
+    <add key="Auth.MicrosoftAccount.ClientId" value="00000000481B784E" />
+    <add key="Auth.MicrosoftAccount.ClientSecret" value="32XRPDH0x9jrz2mDj1rRMJs" />
```
Comment (joelverhagen) [inline src/NuGetGallery/Web.config]: You may want to reset these guys. 👀

**Comment by joelverhagen** [comment]
Could we have a bit of whitespace between the Sign in button and the "No account? Create one!" line? They look crowded.

File: src/NuGetGallery/Controllers/AuthenticationController.cs [hunk 2]
```diff
@@ -69,15 +69,30 @@ public virtual ActionResult LogOn(string returnUrl)
...
-                return SafeRedirect(returnUrl);
+                return LoggedInRedirect(returnUrl);
             }
 
             return SignInView(new LogOnViewModel());
         }
 
         /// <summary>
         /// Sign In\Register view
+        /// <summary>
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/AuthenticationController.cs]: the comment is duplicated

File: src/NuGetGallery/Services/TelemetryService.cs [hunk 3]
```diff
@@ -153,6 +157,28 @@ public void TrackVerifyPackageKeyEvent(string packageId, string packageVersion,
             });
         }
 
+        public void TrackNewUserRegistrationEvent(User user, Credential credential)
+        {
+
+            if (user == null)
```
Comment (skofman1) [inline src/NuGetGallery/Services/TelemetryService.cs]: nit: extra line

File: src/NuGetGallery/Services/TelemetryService.cs [hunk 4]
```diff
@@ -153,6 +157,28 @@ public void TrackVerifyPackageKeyEvent(string packageId, string packageVersion,
...
+
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+
+            TrackEvent(Events.NewUserRegistration, properties => {
+                properties.Add(ClientVersion, GetClientVersion());
+                properties.Add(ProtocolVersion, GetProtocolVersion());
+                properties.Add(ClientInformation, GetClientInformation());
```
Comment (skofman1) [inline src/NuGetGallery/Services/TelemetryService.cs]: we don't need client information for this event. It will always be a browser.

**Comment by skofman1** [comment]
Looks good! A few comments.

File: src/NuGetGallery/Authentication/AuthenticationService.cs [hunk 5]
```diff
@@ -325,6 +287,7 @@ private void InitCredentialFormatters()
             Entities.Users.Add(newUser);
             await Entities.SaveChangesAsync();
 
+            _telemetryService.TrackNewUserRegistrationEvent(newUser, credential);
```
Comment (cristinamanum) [inline src/NuGetGallery/Authentication/AuthenticationService.cs]: Will this log personal data to AI? If yes, would it be possible to move it Geneva audit pipeline?

Comment (shishirx34) [reply src/NuGetGallery/Authentication/AuthenticationService.cs]: no it won't, it simply logs the account creation time and the type of account(password/external.microsoft)

File: src/NuGetGallery/Controllers/AuthenticationController.cs (hunk 2 above)
Comment (shishirx34) [reply src/NuGetGallery/Controllers/AuthenticationController.cs]: ah auto merge

**Pull request merged**